packages/firebase_storage/android/src/main/java/io/flutter/plugins/firebase/storage/FirebaseStoragePlugin.java

```diff
@@ -46,12 +46,36 @@ public void onMethodCall(MethodCall call, final Result result) {
       case "StorageReference#getData":
         getData(call, result);
         break;
+      case "StorageReference#deleteFile":
+        deleteData(call, result);
+        break;
       default:
         result.notImplemented();
         break;
     }
   }
 
+  private void deleteData(MethodCall call, final Result result) {
+    Map<String, String> arguments = (Map<String, String>) call.arguments;
+    String path = arguments.get("path");
+    StorageReference ref = firebaseStorage.getReference().child(path);
+    final Task<Void> deleteTask = ref.delete();
+    deleteTask.addOnSuccessListener(
+        new OnSuccessListener<Void>() {
+          @Override
+          public void onSuccess(Void aVoid) {
+            result.success(true);
```

Comment:
This can be `null`, since this is just confirming that the call to the native side was successful. You can generate the `DeleteTaskResult` object based on success or failure of the invocation no need to pass it in here.